component/constraint/model/SurfacePressureModel.inl

```diff
@@ -183,6 +188,11 @@ void SurfacePressureModel<DataTypes>::bwdInit()
     if(m_componentstate != ComponentState::Valid)
             return ;
 
+
+    d_cavityVolume.setReadOnly(true);
+    d_initialCavityVolume.setReadOnly(true);
+    d_deltaCavityVolume.setReadOnly(true);
+
```

Comment:
Did you detect a malfunction? Is it why you are doing these lines again in the bwdInit()?